hr_worked_days_from_timesheet/models/hr_payslip.py

```diff
@@ -0,0 +1,82 @@
+# -*- coding: utf-8 -*-
+# © 2012 Odoo Canada
+# © 2015 Acysos S.L.
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html)
+
+from openerp import models, fields, api
+from openerp.tools.translate import _
+from openerp.exceptions import Warning as UserError
+
+
+class HrPayslip(models.Model):
+    _inherit = 'hr.payslip'
+
+    @api.multi
+    def _timesheet_mapping(self, timesheet_sheets, payslip, date_from,
+                           date_to):
+        """This function takes timesheet objects imported from the timesheet
+        module and creates a dict of worked days to be created in the payslip.
+        """
+        # Create one worked days record for each timesheet sheet
+        wd_model = self.env['hr.payslip.worked_days']
+        for ts_sheet in timesheet_sheets:
+            # Get formated date from the timesheet sheet
+            date_from_formated = fields.Date.to_string(
+                fields.Datetime.from_string(ts_sheet.date_from))
+            number_of_hours = 0
+            for ts in ts_sheet.timesheet_ids:
+                if date_from <= ts.date <= date_to:
+                    number_of_hours += ts.unit_amount
```

Comment:
@andhit-r Here is should check if the unit of measure is Hours in timesheet and if not to transform it to hours...what if someone record the timesheet in minutes....